### PR TITLE
Finish typing layers.base helper files

### PR DESCRIPTION
**Pull request by dstansby**
# References and relevant issues

# Description
More typing, this time for the layers.base helper files.

### CODE REVIEW

File: napari/layers/base/_base_mouse_bindings.py [hunk 1]
```diff
@@ -155,9 +167,9 @@ def transform_with_box(layer, event):
 
     # we work in data space so we're axis aligned which simplifies calculation
     # same as Layer.data_to_world
-    initial_data_to_world = layer._transforms[1:].simplified.set_slice(
-        event.dims_displayed
-    )
+    simplified = layer._transforms[1:].simplified
+    assert simplified is not None
```
Comment (psobolewskiPhD) [inline napari/layers/base/_base_mouse_bindings.py]: Sorry, naive question: I don't get this assert vs. if/else or even try?

**Review by psobolewskiPhD — APPROVED** [review]

File: napari/layers/base/_base_mouse_bindings.py (hunk 1 above)
Comment (Czaki) [reply napari/layers/base/_base_mouse_bindings.py]: The problem is that for transform chain simplified may return none if transform is empty. 

Personally I prefer to change logic of transform and raise exception when called simplified on empty transform chain

Comment (psobolewskiPhD) [reply napari/layers/base/_base_mouse_bindings.py]: AH, so by asserting we make the type checker happy with the return type?

Comment (dstansby) [reply napari/layers/base/_base_mouse_bindings.py]: Yep, that's right

Comment (Czaki) [reply napari/layers/base/_base_mouse_bindings.py]: I prefer to merge #6674 first and remove this assert

Comment (Czaki) [reply napari/layers/base/_base_mouse_bindings.py]: @dstansby I have merged #6674 so you could update this PR and remove assert.

**Comment by psobolewskiPhD** [comment]
#6674 is merged, but now we have some conflicts 😢

**Comment by dstansby** [comment]
The rebase was just some double quotes conflicting with single quotes, should all be fixed now 👍

**Comment by Czaki** [comment]
@dstansby please remove obsolete now assert

File: napari/layers/base/_base_mouse_bindings.py (hunk 1 above)
Comment (dstansby) [inline napari/layers/base/_base_mouse_bindings.py]:
```suggestion
```

**Review by Czaki — APPROVED** [review]

**Pull request merged**